addons/web_editor/controllers/main.py

```diff
@@ -318,7 +324,17 @@ def _attachment_create(self, name='', data=False, url=False, res_id=False, res_m
         else:
             raise UserError(_("You need to specify either data or url to create an attachment."))
 
-        attachment = request.env['ir.attachment'].create(attachment_data)
+        if request.env.user.has_group('base.group_portal'):
+            request.env[res_model].check_access_rights('write')
+            if res_id:
+                request.env[res_model].browse(res_id).check_access_rule('write')
```

Comment:
Definitely worth a comment in the code to explain this part of the code / those checks